app/services/hyrax/analytics/google.rb

```diff
@@ -0,0 +1,217 @@
+# frozen_string_literal: true
+
+require 'oauth2'
+require 'signet/oauth_2/client'
+
+# OVERRIDE: Hyrax hyrax-v3.5.0 to make analytics a bit more dynamic
+# TODO(alishaevn): remove once https://github.com/samvera/hyrax/pull/6063 is available in our current hyrax version
+
+module Hyrax
+  module Analytics
+    module Google
+      extend ActiveSupport::Concern
+      # rubocop:disable Metrics/BlockLength
+      class_methods do
+        # Loads configuration options from config/analytics.yml for Hyrax apps only. You only need PRIVATE_KEY_PATH or
+        # PRIVATE_KEY_VALUE. VALUE takes precedence.
+        # Expected structure:
+        # `analytics:`
+        # `  google:`
+        # `    app_name: <%= ENV['GOOGLE_OAUTH_APP_NAME']`
+        # `    app_version: <%= ENV['GOOGLE_OAUTH_APP_VERSION']`
+        # `    privkey_path: <%= ENV['GOOGLE_OAUTH_PRIVATE_KEY_PATH']`
+        # `    privkey_value: <%= ENV['GOOGLE_OAUTH_PRIVATE_KEY_VALUE']`
+        # `    privkey_secret: <%= ENV['GOOGLE_OAUTH_PRIVATE_KEY_SECRET']`
+        # `    client_email: <%= ENV['GOOGLE_OAUTH_CLIENT_EMAIL']`
```

Comment:
although I deleted the "analytics.yml" file for this app (as can happen with other hyku apps with this implementation), this file belongs to hyrax. so I left this comment.